src/Microsoft.Health.Dicom.Blob/Features/Storage/BlobFileStore.cs

```diff
@@ -75,6 +77,79 @@ public class BlobFileStore : IFileStore
         }
     }
 
+    /// <inheritdoc />
+    public async Task<Uri> StoreFileInBlocksAsync(
+        long version,
+        Stream stream,
+        IDictionary<string, long> blockLengths,
+        CancellationToken cancellationToken)
+    {
+        EnsureArg.IsNotNull(stream, nameof(stream));
+        EnsureArg.IsNotNull(blockLengths, nameof(blockLengths));
+        EnsureArg.IsGte(blockLengths.Count, 0, nameof(blockLengths.Count));
+
+        BlockBlobClient blobClient = GetInstanceBlockBlobClient(version);
+
+        try
+        {
+            stream.Seek(0, SeekOrigin.Begin);
```

Comment:
I would say it's not up for the method to seek the `Stream`; the caller should do this. There's no guarantee a `Stream` is seekable

Reply:
There seemed to be pattern in other method, we need to change later